File: core/src/main/scala/com/graphbrain/eco/nodes/DummyFun.scala
package com.graphbrain.eco.nodes

import com.graphbrain.eco.NodeType.NodeType
import com.graphbrain.eco.{Context, NodeType}

class DummyFun(val name: String, params: Array[ProgNode], lastTokenPos: Int= -1) extends FunNode(params, lastTokenPos) {
  override val label = name

  override def ntype: NodeType = NodeType.Boolean
}

File: core/src/main/java/com/graphbrain/eco/nodes/DummyFun.java
package com.graphbrain.eco.nodes;

import com.graphbrain.eco.Context;
import com.graphbrain.eco.Contexts;
import com.graphbrain.eco.NodeType;

public class DummyFun extends FunNode {

    private String name;

    public DummyFun(String name, ProgNode[] params, int lastTokenPos) {
        super(params, lastTokenPos);
        this.name = name;
    }

    public DummyFun(String name, ProgNode[] params) {
        this(name, params, -1);
    }

    @Override
    public String label(){return name;}

    @Override
    public NodeType ntype(Context ctxt) {
        return NodeType.Boolean;
    }

    @Override
    public void eval(Contexts ctxts) {

    }

    public String getName() {
        return name;
    }
}